src/tasks/TaskDefinitionBase.ts

```diff
@@ -13,3 +13,17 @@ export interface TaskDefinitionBase extends TaskDefinition {
     label?: string;
     dependsOn?: string[] | DependsOn;
 }
+
+export type DockerLabels = { includeDefaults?: boolean; } & { [key: string]: string; };
+
+export const defaultVsCodeLabels: { [key: string]: string } = {
+    'com.microsoft.created-by': 'visual-studio-code'
+};
+
+export function getAggregateLabels(labels: DockerLabels | undefined, defaultLabels: { [key: string]: string }): { [key: string]: string } {
+    const { includeDefaults, ...explicitLabels } = labels || {};
+
+    return (includeDefaults !== false)

```

Comment:
Just for my education: why `includeDefaults !== false` instead of just `includeDefaults`?

Reply:
`includeDefaults` can be either `undefined`, which is the _real_ default, or `true` or `false`. `undefined` is considered "falsy", so using `if (includeDefaults !== false)` treats `undefined` as `true`, whereas `if (includeDefaults)` treats `undefined` as `false`.

Reply:
[Here](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)'s a really interesting article on falsy-ness, which is definitely important for understanding the behavior of things like the above, along with how `||` and `&&` really behave in the context of assignments.

The wonderful `?.` operator doesn't exist until TypeScript 3.7 (due this month!), so instead of pretty things like `foo?.bar?.baz` we have to do `foo && foo.bar && foo.bar.baz` (which is also slightly different from the former).

Reply:
The intent is `includeDefaults` be treated as `true` if omitted (i.e. if `undefined`), so we test whether it hasn't been set (explicitly) to `false`.

Reply:
In this case I suggest that we change the setting to `omitDefaults` so that we can treat it in code according to established JS conventions around truthiness

Reply:
Personally, I don't like having to set things to `true` in order to turn behavior *off*, which is why I used `includeDefaults` (which defaults to true from the user's perspective).  I'm not sure whether the mechanics of JSON to JS to truthiness conversion matter to the user (or should matter).

Reply:
That's fair